File: backend/src/main/scala/Match.scala
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

case class Match(id: Int, audioStart: Double, audioEnd: Double, textStart: Double, textEnd: Double) {
}
object Match {
  implicit val encoder = deriveEncoder[Match]
  implicit val decoder = deriveDecoder[Match]
}

File: frontend/src/app/models/match.ts
export class Match {
  id: number;
  audioStart: number;
  audioEnd: number;
  textStart: number;
  textEnd: number;

  constructor(id: number, audioStart: number, audioEnd: number, textStart: number, textEnd: number) {
    this.id = id;
    this.audioStart = audioStart;
    this.audioEnd = audioEnd;
    this.textStart = textStart;
    this.textEnd = textEnd;
  }
}
